src/main/java/com/conveyal/r5/analyst/cluster/PathWriter.java

```diff
@@ -61,14 +64,27 @@ public void recordPathsForTarget (int[] paths) {
             // Write the path indexes used to reach this target at each iteration, delta coded within each target.
             int prev = 0;
             for (int pathIdx : paths) {
-                outputStream.write(pathIdx - prev);
+                writeInt(pathIdx - prev);
                 prev = pathIdx;
             }
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
 
+    /**
+     *
+     * @param value
+     * @throws IOException
+     */
+    private void writeInt (int value) throws IOException {
```

Comment:
This method definitely needs a comment explaining what it does and why. It seems strange to re-implement a fundamental method to write out an integer (more code we have to maintain ourselves). It also looks to me like it's writing out a big-endian representation of an int, which is the opposite of what we want. I think what you're looking for is DataOutputStream, which wraps an OutputStream to write more complex data types than bytes. But it's permanently big-endian, so for these cases we use LittleEndianDataOutputStream.

Also style-wise we avoid javadoc `@param` etc. unless we are writing a full description. Also just style-wise, this function should be declared static because its work does not depend on instance state.